phobos/blender/io/blender2phobos.py

```diff
@@ -660,8 +660,14 @@ def deriveSubmechanism(obj, logging=False):
         if k not in reserved_keys.INTERNAL_KEYS+reserved_keys.SUBMECHANISM_KEYS
     }
     for prop in reserved_keys.SUBMECHANISM_KEYS:
-        jointIDs = obj[prop]
-        values[prop] = jointIDs
+        if type(obj[prop]) == str:
+            values[prop] = obj[prop]
+        elif type(obj[prop]) == dict:
+            values[prop] = {}
+            for key, j in obj[prop]:
+                values[prop][key] = j.get("joint/name",j.name)
+        elif prop is not None:
```

Comment:
This case is a bit unclear for two reasons:
1. prop should never be None as it comes from a hardcoded list of keys
2. This is only active for the jointnames_* entries.
To align it with the other's I'd change it to `elif type(obj[prop]) == list or "Array" in repr(type(obj[prop])):`